flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -400,6 +400,24 @@ void createIcebergTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
     }
   }
 
+  // skip the AbstractConstraint comparison
+  public boolean schemaCompare(org.apache.flink.table.api.Schema rawSchema,
+                               org.apache.flink.table.api.Schema newSchema) {
+
+    org.apache.flink.table.api.Schema.UnresolvedPrimaryKey rawPK = rawSchema.getPrimaryKey().orElse(null);
+
+    org.apache.flink.table.api.Schema.UnresolvedPrimaryKey newPK = newSchema.getPrimaryKey().orElse(null);
+    if ((rawPK == null && newPK != null) || (rawPK != null && newPK == null)) {
+      return false;
+    }
+    boolean exp = rawSchema.getColumns().equals(newSchema.getColumns()) &&
+            rawSchema.getWatermarkSpecs().equals(newSchema.getWatermarkSpecs());
+    boolean exp2 = rawPK != null && newPK != null ?
+            rawPK.getColumnNames().equals(newPK.getColumnNames())
+            : true;
+    return exp && exp2;
```

Comment:
Rather than having generic names like `exp` and `exp2`, can you either use real names or just return early when the columns or watermarks aren't equal?